src/abstractdataframe/selection.jl

```diff
@@ -376,12 +376,60 @@ _transformation_helper(df::AbstractDataFrame, col_idx::Int, (fun,)::Ref{Any}) =
 
 _empty_astable_helper(fun, len) = [fun(NamedTuple()) for _ in 1:len]
 
+@noinline _table_transformation(df_sel::AbstractDataFrame, fun) = fun(Tables.columntable(df_sel))
+
+function _sum_fast(cols::Vector{<:AbstractVector})
+    isempty(cols) && throw(ArgumentError("No columns selected for reduction"))
+    return sum(cols)
+end
+
+function _sum_skipmissing_fast(cols::Vector{<:AbstractVector})
+    isempty(cols) && throw(ArgumentError("No columns selected for reduction"))
+    sumz = nothing
+    for col in cols
+        try
+            zi = zero(eltype(col))
+            if isnothing(sumz)
+                sumz = zi
+            elseif !ismissing(zi) # zi is missing if eltype is Missing
+                sumz += zi
+            end
+        catch e
+            if e isa MethodError && e.f === Base.zero
+                sumz = nothing
+                break
+            else
+                throw(e)
+            end
+        end
+    end
+    # this will happen if eltype of some columns do not support zero
+    # or all columns have eltype Missing
+    isnothing(sumz) && return nothing
+    init = fill(sumz, length(cols[1]))
+
+    return foldl(cols, init=init) do l, r
+        l .= ifelse.(ismissing.(r), l, l .+ r)
+    end
+end
+
 function _transformation_helper(df::AbstractDataFrame, col_idx::AsTable, (fun,)::Ref{Any})
-    tbl = Tables.columntable(select(df, col_idx.cols, copycols=false))
-    if isempty(tbl) && fun isa ByRow
+    df_sel = select(df, col_idx.cols, copycols=false)
+    if ncol(df_sel) == 0 && fun isa ByRow
         return _empty_astable_helper(fun.fun, nrow(df))
+    elseif typeof(fun) === typeof(Base.sum) || typeof(fun) === ByRow{typeof(Base.sum)}
+        return _sum_fast(map(identity, eachcol(df_sel)))
+    elseif typeof(fun) === ByRow{typeof(Base.:∘(Base.sum, Base.skipmissing))}
+        fastsum = _sum_skipmissing_fast(map(identity, eachcol(df_sel)))
+        if isnothing(fastsum)
+            slowsum = _table_transformation(df_sel, fun)
+            T = mapreduce(typeof, promote_type, slowsum)
```

Comment:
Maybe this could be slightly faster?
```suggestion
            T = isconcretetype(nonmissingtype(T)) ? T : mapreduce(typeof, promote_type, slowsum)
```

Reply:
I am not sure what you mean here. `T` is undefined before this line AFAICT.

Reply:
OK - I see what you mean, but it needs another syntax.